app/models/payment/braintree_payment_method.rb

```diff
@@ -1,3 +1,4 @@
 class Payment::BraintreePaymentMethod < ActiveRecord::Base
-  belongs_to :customer, class_name: Payment::BraintreeCustomer, foreign_key: :customer_id
+  belongs_to :customer,     class_name: Payment::BraintreeCustomer
```

Comment:
yes - I was struggling so decided to drop closer to rails' conventions to keep things a wee bit simpler.